Add Llama2 inference benchmark under a new "benchmarks" section (#435)
huggingface/optimum-neuron: docs/source/benchmarks/inferentia-llama2.mdx

<!---
Copyright 2024 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

> # Llama-2 models on AWS Inferentia2

Reviewer: ```suggestion
# Llama performance on AWS Inferentia2 (Latency & Througput)
```

> How much efficient is text-generation on Inferentia2?  Let's figure out!

Reviewer: ```suggestion
How fast is Llama on Inferentia2?  Let's figure out!
```

For this benchmark we will use the LLama 2 7B and 13B models with different configurations:

| Model type                 | num cores | batch_size |
|----------------------------|-----------|------------|
| Llama2 7B - L (latency)    | 24        | 1          |
| Llama2 7B - T (throughput) | 24        | 4          |
| Llama2 13B - L (latency)   | 24        | 1          |
| Llama2 13B - T (throughput)| 24        | 4          |

*Note: all models are compiled with a maximum sequence length of 2048.*

All models are compiled to use the full extent of cores available on the `inf2.48xlarge` instance.

*Note: please refer to the [inferentia2 product page](https://aws.amazon.com/ec2/instance-types/inf2/) for details on the available instances.*

> We created two "latency" oriented configurations for the `llama2 7B` and `llama2 13B` models that can serve only one request at a time, but at full speed.
> 
> We also created two "throughput" oriented configurations to serve up to four requests in parallel.

Reviewer: ```suggestion
We created two "latency" oriented configurations for the `llama2 7B` and `llama2 13B` models that can serve only one request at a time, but at full speed and two "throughput" oriented configurations to serve up to four requests in parallel.
```

To evaluate the models, we generate tokens up to a total sequence length of 1024, starting from
256 input tokens (i.e. we generate 256, 512 and 768 tokens).

> ### Encoding time

Reviewer: ```suggestion
## Encoding time (time to first token)
```

> The encoding time is the time required to process the input tokens and generate the first output token.

Reviewer: ```suggestion
The encoding time or time to first token is the time required to process the input tokens and generate the first output token.
```

It is a very important metric, as it corresponds to the latency directly perceived by the user when streaming generated tokens.

We test the encoding time for increasing context sizes, 256 input tokens corresponding roughly to a typical Q/A usage,
while 768 is more typical of a Retrieval Augmented Generation (RAG) use-case.

Encoding time is expressed in **seconds**.

![Llama2 inferentia2 encoding-time](https://raw.githubusercontent.com/huggingface/optimum-neuron/main/docs/assets/benchmarks/inferentia-llama2/encoding-times.png "Encoding time")

We can see that all deployed models exhibit excellent response times, even for long contexts.

> ### End-to-end Latency

Reviewer: ```suggestion
## End-to-end Latency
```
